crates/solver/src/main.rs

```diff
@@ -422,6 +434,8 @@ async fn main() {
                 chain_id,
                 web3.clone(),
                 token_info_fetcher.clone(),
+                args.balancer_factories
+                    .unwrap_or_else(BalancerFactoryKind::all),
```

Comment:
Never mind, I see we use all if not specified. If possible, this might be more easily interpreted if this were set as the default parameter... Doesn't the runtime fail if parameters without defaults aren't specified?

Reply:
We do the same for `baseline-sources`. This allows the default to be dynamic. I mainly use this so that the default is the result of some function evaluation (instead of a string literal).